applications/RomApplication/costum_strategies/rom_builder_and_solver.h

```diff
@@ -0,0 +1,811 @@
+			for (unsigned int i = 0; i < tmp.size(); ++i) 
+			{
+			rX[node_aux_id*mNodalDofs + i] = tmp[i];
+			}
+        }
+    }
+
+    void GetDofValues(const std::vector<DofPointerType>& rDofList, TSystemVectorType& rX)
+    {
+        unsigned int i=0;
+        for(auto& dof : rDofList)
+            rX[i++] = dof->GetSolutionStepValue();
+    }
+
+    /*@{ */
+
+    /**
+            Function to perform the building and solving phase at the same time.
+            It is ideally the fastest and safer function to use when it is possible to solve
+            just after building
+     */
+    virtual void BuildAndSolve(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& A,
+        TSystemVectorType& Dx,
+        TSystemVectorType& b) override
+	{
+		
+        //define a dense matrix to hold the reduced problem
+        Matrix Arom = ZeroMatrix(mRomDofs,mRomDofs);
+        Vector brom = ZeroVector(mRomDofs);
+        TSystemVectorType x(Dx.size());
+
+        //find the rom basis
+        this->GetDofValues(mDofList,x);
+
+		double start_build4 = OpenMPUtils::GetCurrentTime();        
+		Vector xrom = this->ProjectToReducedBasis(x, rModelPart.Nodes());
+		const double stop_build4 = OpenMPUtils::GetCurrentTime();
+		KRATOS_INFO_IF("ROMBuilderAndSolver", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Project to reduced basis time: " << stop_build4 - start_build4 << std::endl;
+
+		        
+        //build the system matrix by looping over elements and conditions and assembling to A
+        KRATOS_ERROR_IF(!pScheme) << "No scheme provided!" << std::endl;
+
+        // Getting the elements from the model
+        const int nelements = static_cast<int>(rModelPart.Elements().size());
+
+        // Getting the array of the conditions
+        const int nconditions = static_cast<int>(rModelPart.Conditions().size());
+
+        ProcessInfo& CurrentProcessInfo = rModelPart.GetProcessInfo();
+        ModelPart::ElementsContainerType::iterator el_begin = rModelPart.ElementsBegin();
+        ModelPart::ConditionsContainerType::iterator cond_begin = rModelPart.ConditionsBegin();
+
+        //contributions to the system
+        LocalSystemMatrixType LHS_Contribution = LocalSystemMatrixType(0, 0);
+        LocalSystemVectorType RHS_Contribution = LocalSystemVectorType(0);
+
+        //vector containing the localization in the system of the different
+        //terms
+        Element::EquationIdVectorType EquationId;
+
+        // assemble all elements
+        double start_build = OpenMPUtils::GetCurrentTime();
+
+//       #pragma omp parallel firstprivate(nelements,nconditions, LHS_Contribution, RHS_Contribution, EquationId )
+        {
+//            # pragma omp for  schedule(guided, 512) nowait
+            for (int k = 0; k < nelements; k++)
+            {
+                ModelPart::ElementsContainerType::iterator it = el_begin + k;
```

Comment:
```suggestion
               auto it = el_begin + k;
```
I also suggest to change the iterator name to `it_elem`.